src/libsync/propagateremotemkdir.cpp

```diff
@@ -131,15 +134,57 @@ void PropagateRemoteMkdir::success()
     itemCopy._etag.clear();
 
     // save the file id already so we can detect rename or remove
+    Q_ASSERT(thread() == qApp->thread());
     const auto result = propagator()->updateMetadata(itemCopy);
     if (!result) {
         done(SyncFileItem::FatalError, tr("Error writing metadata to the database: %1").arg(result.error()));
         return;
-    } else if (result.get() == Vfs::ConvertToPlaceholderResult::Locked) {
-        done(SyncFileItem::FatalError, tr("The file %1 is currently in use").arg(_item->_file));
+    }
+#ifdef Q_OS_WIN
+    else if (result.get() == Vfs::ConvertToPlaceholderResult::Locked) {
+        // updateMetadata invokes convertToPlaceholder
+        // On Windows convertToPlaceholder can be blocked by the OS
+        // in case the file lock is active.
+        // Retry the conversion for UpdateMetaDataRetyTimeOut
+        // TODO: make update updateMetadata async in general
+
+        // QtConcurrent does not support moves
+        // use shared_ptr to manage the Result object
+        using resultType = decltype(result);
+        using resultTypePtr = decltype(std::shared_ptr<resultType>());
+
+        auto *watcher = new QFutureWatcher<resultTypePtr>(this);
+        connect(watcher, &QFutureWatcherBase::finished, this, [watcher, this] {
+            const auto *result = watcher->result().get();
+            if (*result) {
+                Q_ASSERT(result->get() == Vfs::ConvertToPlaceholderResult::Ok);
+                done(SyncFileItem::Success);
+            } else {
+                done(SyncFileItem::SoftError, result->error());
+            }
+        });
+
+        watcher->setFuture(QtConcurrent::run([p = propagator(), itemCopy] {
+            // Try to update the meta data with a 30s timeout
+            const auto start = std::chrono::steady_clock::now();
+            while ((std::chrono::steady_clock::now() - start) < UpdateMetaDataRetyTimeOut) {
```

Comment:
I like `QElapsedTimer` for these kind of things.

Reply:
Hm but then we have to either convert units or use std::chrono for the conversion and in the end we don't gain much  besides more involved components.